tests/models/test_NaiveBayes.py

```diff
@@ -0,0 +1,123 @@
+import shutil
+import tempfile
+import unittest
+
+import numpy as np
+from sklearn.model_selection import train_test_split
+import scipy.stats
+
+from pymc3_models import GaussianNaiveBayes
+from pymc3_models.exc import PyMC3ModelsError
+
+
+class GaussianNaiveBayesTestCase(unittest.TestCase):
+    def setUp(self):
+        """
+        Set up a test case with synthetic data.
+        """
+
+        self.num_cats = 3
+        self.num_pred = 10
+        self.num_samples = 100000
+
+        # Generate data
+        ## Priors
+        self.alpha = np.ones(self.num_cats)
+        self.pi = np.random.dirichlet(self.alpha)
+        self.mu = np.random.normal(0, 100, size=(self.num_cats, self.num_pred))
+        self.sigma = scipy.stats.halfnorm(loc=0, scale=100).rvs(size=(self.num_cats, self.num_pred))
+        ## Data
+        Y = np.random.choice(range(self.num_cats), self.num_samples, p=self.pi)
+        x_vectors = []
+        for i in Y:
+            x_vectors.append(np.random.normal(self.mu[i], self.sigma[i]))
+        X = np.vstack(x_vectors)
+
+        # Split into train/test sets
+        split = train_test_split(X, Y, test_size=0.4)
+        self.X_train = split[0]
+        self.X_test = split[1]
+        self.Y_train = split[2]
+        self.Y_test = split[3]
+
+        self.num_training_samples = self.Y_train.shape[0]
+        self.test_GNB = GaussianNaiveBayes()
+        self.test_dir = tempfile.mkdtemp()
+
+        # Fit the model once and for all
+        self.test_GNB.fit(self.X_train, self.Y_train, minibatch_size=2000)
+
+    def tearDown(self):
+        """
+        Tear down the testing environment.
+        """
+        shutil.rmtree(self.test_dir)
+
+
+class GaussianNaiveBayesFitTestCase(GaussianNaiveBayesTestCase):
+    def test_fit_returns_correct_model(self):
+        """
+        Test the model initialization and fit.
+
+        Currently, only the sign of inferred parameters is checked
+        against the sign of the parameters used to generate the data.
+
+        TOOO: Find better strategies to test probabilistic code.
+        """
+
+        print('')
+
+        # Check that the model correctly infers dimensions
+        self.assertEqual(self.num_cats, self.test_GNB.num_cats)
+        self.assertEqual(self.num_training_samples, self.test_GNB.num_training_samples)
+        self.assertEqual(self.num_pred, self.test_GNB.num_pred)
+
+        # TODO: How do you write tests for a stochastic mode?
+        # TODO: Diagnose the sampling with a reasonable sampling size?
+        np.testing.assert_equal(
```

Comment:
These long lines are inconsistent about where the closing paren is. 
I would prefer: 
```
np.testing.assert_equal(
            np.sign(self.pi),
            np.sign(self.test_GNB.trace['pi'].mean(axis=0))
)
```

Reply:
:+1: